python/lsst/multiprofit/fitting/fit_bootstrap_model.py

```diff
@@ -0,0 +1,269 @@
+            n_rows=self.config_boot.observation.n_rows,
+            n_cols=self.config_boot.observation.n_cols,
+        )
+        return image.data
+
+    def get_catalog(self) -> astropy.table.Table:
+        return self.config_boot.catalog
+
+    def get_psf_image(
+        self, source: astropy.table.Row | Mapping[str, Any], config: CatalogPsfFitterConfig | None = None
+    ) -> np.ndarray:
+        rng = np.random.default_rng(source["id"])
+        image = self.image
+        config_obs = self.config_boot.observation
+        return image + rng.standard_normal(image.shape) * np.sqrt(
+            (image + config_obs.background) / config_obs.gain
+        )
+
+    def model_post_init(self, __context):
+        self.config_boot.freeze()
+
+
+class CatalogExposureSourcesBootstrap(CatalogExposureSourcesABC, pydantic.BaseModel):
+    """A CatalogExposure for bootstrap fitting of source catalogs."""
+
+    model_config = frozen_arbitrary_allowed_config
+
+    config_boot: CatalogSourceBootstrapConfig = pydantic.Field(
+        title="A CatalogSourceBootstrapConfig to be frozen"
+    )
+    table_psf_fits: astropy.table.Table = pydantic.Field(title="PSF fit parameters for the catalog")
+
+    @cached_property
+    def channel(self):
+        channel = g2f.Channel.get(self.config_boot.observation.band)
+        return channel
+
+    def get_catalog(self) -> astropy.table.Table:
+        return self.config_boot.catalog
+
+    def get_psf_model(self, params: Mapping[str, Any]) -> g2f.PsfModel:
+        psf_model = self.psf_model_data.psf_model
+        self.psf_model_data.init_psf_model(self.table_psf_fits[params["id"]])
+        return psf_model
+
+    def get_source_observation(self, source: Mapping[str, Any]) -> g2f.ObservationD:
+        obs = self.config_boot.observation.make_observation()
+        return obs
+
+    def model_post_init(self, __context):
+        config_dict = self.table_psf_fits.meta["config"]
+        config = CatalogPsfFitterConfig()
+        set_config_from_dict(config, config_dict)
+        config_data = CatalogPsfFitterConfigData(config=config)
+        object.__setattr__(self, "psf_model_data", config_data)
+
+
+class CatalogSourceFitterBootstrap(CatalogSourceFitterABC, pydantic.BaseModel):
+    """A catalog fitter that bootstraps a single model.
+
+    This fitter generates a different noisy image of the specified model for
+    each row. The resulting catalog can be used to examine performance and
+    statistics of the best-fit parameters.
+    """
+
+    model_config = frozen_arbitrary_allowed_config
+
+    def get_model_radec(self, source: Mapping[str, Any], cen_x: float, cen_y: float) -> tuple[float, float]:
+        return float(cen_x), float(cen_y)
+
+    def initialize_model(
+        self,
+        model: g2f.ModelD,
+        source: Mapping[str, Any],
+        catexps: list[CatalogExposureSourcesABC],
+        values_init: Mapping[g2f.ParameterD, float] | None = None,
+        centroid_pixel_offset: float = 0,
```

Comment:
`centroid_pixel_offset` is never used

Reply:
This is another overridden function; it's fine to not use it. Actually, I had to add the missing kwargs (which are also unused).